Training/2.0 Libraries.md

```diff
@@ -24,16 +24,25 @@ When to use which? Each of these libraries has very useful functionality but you
 Make a two-dimensional numpy array. First row will be: [1,2,3] and second row will be [4,5,6]. 
 Using numpy functionality find the mean, minimum and max value of your array. Index the maximum value and assign it to a variable named max. Index the minimum value and assign it to a variable named min. 
 
-```
+``` np2d = np.array([[1,2,3][4,5,6]])
+``` max = (np2d.max())
+``` min = (np2d.min())
+``` average = (np2d.mean())
+``` max = 6
+``` min = 1
+``` average = 3.5
 
 ```
 
 ### Task 2b)
 Calculate the trimmed mean of the following array using ONLY Numpy and python's base functions: [67,67,65,50,69,68,67,70,69,66]. The trimmed mean is the mean of the data with the largest and the smallest values (outliers) removed.
 
-```
+```values = [67,67,65,50,69,68,67,70,69,66]
+sort = sorted(values)
+nums = sort[1:8]
+travg = print(sum(nums)/7)
 
-```
+``` trimmed mean = 67
 
```

Comment:
values = [67,67,65,50,69,68,67,70,69,66]  -- this is a list. To create an array, you should use np.array:
values = np.array([67,67,65,50,69,68,67,70,69,66])

Also, by having nums = sort[1:8] rather than [1:9], you are trimming the two largest numbers rather than the largest one only. A better way to do this would have been using the same methods as in part a) to remove the max and min, and then take the average without them:
sort = sorted(values)
nums = sort[1:(len(values) - 1)]     -- this removes the min and max numbers of the array
travg = print(sum(nums)/(len(values)-2))

Reply:
Using the sorted() function and sticking with list manipulation is a clever heuristic approach, but as Alejandra has pointed out, you have to be careful with slicing indexes.
Pro Tip: Python understands reverse iteration, so you can also use `nums = sort[1:-1]`